TerriaJS/terriajs
# Only add groups to `CatalogIndex` if they aren't empty

### Only add groups to `CatalogIndex` if they aren't empty

Fixes https://github.com/TerriaJS/nationalmap/issues/1184

This PR will remove empty groups from generated catalog index files

### To test

I added a group called "Some empty group" to `"simple.json"` and ran the `generateCatalogIndex` script before and after the changes

- **Before** http://ci.terria.io/main/#configUrl=https://gist.githubusercontent.com/nf-s/64e8333b1536bafd3c282352b83b63b0/raw/56651e099344fae52c1f1d6bf072823a6d670d1d/catalog-index-simple-config.json
- **After** http://ci.terria.io/main/#configUrl=https://gist.githubusercontent.com/nf-s/edf1a4e8edc84ab71b2e3e4561077448/raw/a48e1393d119fa2c6f9c0670e2e6b33a7e7fdcee/catalog-index-no-empty-group-config.json
- Open catalog
- Search "Some empty group"
- Note that group shows in **before** link - but doesn't show in **after** link

### Checklist
